src/main/java/browserview/BrowserComponent.java

```diff
@@ -530,6 +567,20 @@ public void switchToTab(String tabName) {
         }
         return Optional.empty();
     }
+
+    public void waitUntilDiscussionPageLoaded() {
+        for (int i = 0; i < WAIT_PAGE_LOAD_MAX_RETRY; i++) {
+            if (!isCurrentUrlDiscussion()) {
+                try {
+                    Thread.sleep(WAIT_PAGE_LOAD_SLEEP_TIME);
+                } catch (IllegalArgumentException | InterruptedException e) {
+                    logger.warn(e);
+                }
+            } else {
+                return;
+            }
```

Comment:
I changed this as requested by @acruis to put the if clause into the for
condition.
On Mar 18, 2016 12:06 PM, "Damith C. Rajapakse" <notifications@github.com>
wrote:

> In src/main/java/browserview/BrowserComponent.java
> <https://github.com/HubTurbo/HubTurbo/pull/1367#discussion_r56611336>:
>
> > @@ -530,6 +567,20 @@ public void switchToTab(String tabName) {
> >          }
> >          return Optional.empty();
> >      }
> > +
> > +    public void waitUntilDiscussionPageLoaded() {
> > +        for (int i = 0; i < WAIT_PAGE_LOAD_MAX_RETRY; i++) {
> > +            if (!isCurrentUrlDiscussion()) {
> > +                try {
> > +                    Thread.sleep(WAIT_PAGE_LOAD_SLEEP_TIME);
> > +                } catch (IllegalArgumentException | InterruptedException e) {
> > +                    logger.warn(e);
> > +                }
> > +            } else {
> > +                return;
> > +            }
>
> Use guard clauses to reduce nesting. It's not a good sign when a big chunk
> of code is nested.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/HubTurbo/HubTurbo/pull/1367/files/a59edc017adf0a9ff9d35d688537e6dcaaf623e4#r56611336>
>